src/reanimated2/WorkletEventHandler.ts

```diff
@@ -20,64 +23,121 @@ function jsListener<Event extends object>(
   };
 }
 
-export default class WorkletEventHandler<Event extends object> {
-  worklet: (event: ReanimatedEvent<Event>) => void;
+class WorkletEventHandlerNative<Event extends object>
+  implements IWorkletEventHandler<Event>
+{
+  eventNames: string[];
+  #worklet: (event: ReanimatedEvent<Event>) => void;
+  #viewTags: Set<number>;
+  #registrations: Map<number, number[]>; // keys are viewTags, values are arrays of registration ID's for each viewTag
+  constructor(
+    worklet: (event: ReanimatedEvent<Event>) => void,
+    eventNames: string[] = []
+  ) {
+    this.#worklet = worklet;
```

Comment:
Even though it's private API we allow the access to `worklet` for Reanimated users since for now there's no other reliable way to merge our event handlers. Let's make it public and we'll make it private once we add support for handler merging.

Reply:
Sure thing